lib/graphql/schema.rb

```diff
@@ -349,13 +349,9 @@ def lazy?(obj)
       !!lazy_method_name(obj)
     end
 
-    # @return [Array<#call>] Middlewares suitable for MiddlewareChain, applied to fields during execution
+    # @return [MiddlewareChain] MiddlewareChain which is applied to fields during execution
     def middleware
-      @middleware ||= if @rescue_middleware
-        [@rescue_middleware].concat(@user_middleware)
-      else
-        @user_middleware
-      end
+      @user_middleware
```

Comment:
I remember this song and dance had something to do with `#redefine`, making sure that middlewares get duped correctly. 

Do you know of this handles in properly? (I can't remember if I wrote a proper test for it 😬 , I'll take a look tonight!)